solr/core/src/java/org/apache/solr/filestore/NodeFileStore.java

```diff
@@ -76,135 +66,54 @@ public SolrJerseyResponse getFile(String path, Boolean sync, String getFrom, Boo
-                }
-                log.info("downloaded file: {}", pathCopy);
-              });
+      ClusterFileStore.pullFileFromNode(coreContainer, fileStore, pathCopy, getFrom);
       return response;
     }
 
     FileStore.FileType type = fileStore.getType(path, false);
-    if (type == FileStore.FileType.NOFILE) {
-      final var fileMissingResponse =
-          instantiateJerseyResponse(FileStoreDirectoryListingResponse.class);
-      fileMissingResponse.files = Collections.singletonMap(path, null);
-      return fileMissingResponse;
-    }
-    if (type == FileStore.FileType.DIRECTORY) {
-      final var directoryListingResponse =
-          instantiateJerseyResponse(FileStoreDirectoryListingResponse.class);
-      final var directoryContents =
-          fileStore.list(path, null).stream()
-              .map(details -> convertToResponse(details))
-              .collect(Collectors.toList());
-      directoryListingResponse.files = Collections.singletonMap(path, directoryContents);
-      return directoryListingResponse;
+    if (type == FileStore.FileType.NOFILE
+        || type == FileStore.FileType.DIRECTORY
+        || (type == FileStore.FileType.FILE && Boolean.TRUE.equals(meta))) {
+      return ClusterFileStore.getMetadata(type, path, fileStore);
     }
-    if (Boolean.TRUE.equals(meta)) {
-      if (type == FileStore.FileType.FILE) {
-        int idx = path.lastIndexOf('/');
-        String fileName = path.substring(idx + 1);
-        String parentPath = path.substring(0, path.lastIndexOf('/'));
-        List<FileStore.FileDetails> l = fileStore.list(parentPath, s -> s.equals(fileName));
 
-        final var fileMetaResponse =
-            instantiateJerseyResponse(FileStoreDirectoryListingResponse.class);
-        fileMetaResponse.files =
-            Collections.singletonMap(path, l.isEmpty() ? null : convertToResponse(l.get(0)));
-        return fileMetaResponse;
-      }
-    } else { // User wants to get the "raw" file
-      // TODO Should we be trying to json-ify otherwise "raw" files in this way?  It seems like a
-      // pretty sketchy idea, esp. for code with very little test coverage.  Consider removing
-      if ("json".equals(req.getParams().get(CommonParams.WT))) {
-        final var jsonResponse = instantiateJerseyResponse(FileStoreJsonFileResponse.class);
-        try {
-          fileStore.get(
-              pathCopy,
-              it -> {
-                try {
-                  InputStream inputStream = it.getInputStream();
-                  if (inputStream != null) {
-                    jsonResponse.response = new String(inputStream.readAllBytes(), UTF_8);
-                  }
-                } catch (IOException e) {
-                  throw new SolrException(
-                      SolrException.ErrorCode.SERVER_ERROR, "Error reading file " + pathCopy);
+    // User wants to get the "raw" file
+    // TODO Should we be trying to json-ify otherwise "raw" files in this way?  It seems like a
```

Comment:
What is the use case FOR this?   Do we have one right now?

Reply:
Not that I know of.  I'm going to nuke it based on the discussion [here](https://github.com/apache/solr/pull/3047/files#r1929771967), and that might flush out something I don't understand.

(We'll need to make sure we're only deprecating it in the 9.x backport, but that's not a big deal...)

Reply:
There's no discussion of it either in the PR or the JIRA that introduced it (unless I'm missing something?).  My guess is that it was an attempt at convenience?

In any case; I think this entire file can now go.  See my comment [above](https://github.com/apache/solr/pull/3047/files#r1929771967) for more details.